libs/cycle-finder/src/max_cycle.cpp

```diff
@@ -113,10 +114,36 @@ void MaxCycle::unblockVertex(vertex v) {
 }
 
 void MaxCycle::filterMaxCycles() {
-
     for (const auto& cycle : cycles_) {
         if (cycle.size() == maxCycleSize_) maxCycles_.push_back(cycle);
     }
 }
 
+void MaxCycle::filterMaxCyclesExact() {
+    // create graphs from all cycles.
+    this->filterMaxCycles();
+    auto graphCycles = std::vector<core::Multigraph>(maxCycles_.size());
+    for (int i = 0; i < graphCycles.size(); i++) {
+        graphCycles[i] = baseMultiGraph_.inducedSubgraph(maxCycles_[i]);
```

Comment:
In the exact algorithm `maxCycles_[i]` contains the same vertex at the beginning and at the end. As a result the method `inducedSubgraph` returns a graph with one additional incorrect vertex, and therefore, the size of the cycle is incorrectly printed.

Reply:
I omitted matrix representation for approximation, because it returns set, not the cycle, I talked about it with Homenda on thursday.